src/game/client/ingameui.cpp

```diff
@@ -13,35 +13,1131 @@
+        }
+
+        if (m_popupMessageData->m_pause) {
+            g_theGameLogic->Set_Game_Paused(false, m_popupMessageData->m_pauseMusic);
+        }
+
+        m_popupMessageData->Delete_Instance();
+        m_popupMessageData = nullptr;
+    }
+}
+
+void InGameUI::Clear_World_Animations()
+{
+    for (auto it = m_worldAnimations.begin(); it != m_worldAnimations.end(); it++) {
+        WorldAnimationData *anim = *it;
+
+        if (anim != nullptr) {
+            anim->m_anim->Delete_Instance();
+            delete anim;
+        }
+
+        m_worldAnimations.erase(it);
+    }
+}
+
+void InGameUI::Post_Draw()
+{
+    if (m_messagesOn) {
+        int message_x = m_messagePosition.x;
+        int message_y = m_messagePosition.y;
+
+        for (int i = MAX_UI_MESSAGES - 1; i >= 0; i--) {
+            if (m_uiMessages[i].display_string != nullptr) {
+                unsigned char red;
+                unsigned char green;
+                unsigned char blue;
+                unsigned char alpha;
+                Get_Color_Components(m_uiMessages[i].color, &red, &green, &blue, &alpha);
+                m_uiMessages[i].display_string->Draw(
+                    message_x, message_y, m_uiMessages[i].color, Make_Color(0, 0, 0, alpha));
+                message_y += m_uiMessages[i].display_string->Get_Font()->m_height;
+            }
+        }
+    }
+
+    if (m_militarySubtitle != nullptr) {
+        ICoord2D position = m_militarySubtitle->position;
+        unsigned char red;
+        unsigned char green;
+        unsigned char blue;
+        unsigned char alpha;
+        Get_Color_Components(m_militarySubtitle->color, &red, &green, &blue, &alpha);
+        unsigned int border_color = Make_Color(0, 0, 0, alpha);
+
+        for (unsigned int j = 0; j <= m_militarySubtitle->current_display_string; j++) {
+            m_militarySubtitle->display_strings[j]->Draw(position.x, position.y, m_militarySubtitle->color, border_color);
+            int height;
+            m_militarySubtitle->display_strings[j]->Get_Size(nullptr, &height);
+            position.y += height;
+        }
+
+        if (m_militarySubtitle->block_drawn) {
+            int height =
+                m_militarySubtitle->display_strings[m_militarySubtitle->current_display_string]->Get_Font()->m_height;
+            int width = height * 0.8f;
+            g_theDisplay->Draw_Fill_Rect(
+                m_militarySubtitle->block_pos.x, m_militarySubtitle->block_pos.y, width, height, m_militarySubtitle->color);
+        }
+    }
+
+    if (g_theGameLogic->Get_Frame() != 0) {
+        int superweapon_x = g_theDisplay->Get_Width() * m_superweaponPosition.x;
+        int superweapon_y = g_theDisplay->Get_Height() * m_superweaponPosition.y;
+        int view_height = g_theTacticalView->Get_Height() * 0.82f;
+        bool draw_limit_reached = false;
+
+        for (int k = 0; k < MAX_PLAYER_COUNT; k++) {
+            int border_color = Make_Color(0, 0, 0, 255);

```

Comment:
This can be constexpr because `Make_Color` is constexpr and all inputs are constants.